packages/dev/deploy/deploy.ts

```diff
@@ -25,86 +25,93 @@ interface DeploymentConfig {
-  const setField = async function (contract: string, getFunc: string, setFunc: string, val: any, options = {
+/** helper: set a field on a contract only if it was changed.
+ * the "deploy" mechanism has the property of "re-deploy only on change". this method replicate the logic for calling a setter.
+ * @param deployments - hardhat deploy extension
+ * @param contract - a contract deployed earlier using "deploy"
+ * @param getFunc - a getter (with no params) to read current value
+ * @param setFunc - a setter function (accepting a single value) to set the new value
+ * @param val - the value we want the field to have
+ * @param deployer
+ */
+async function setField (deployments: DeploymentsExtension, contract: string, getFunc: string, setFunc: string, val: string, deployer: string): Promise<void> {
+  const options: TxOptions = {
     from: deployer,
     log: true
-  }): Promise<void> {
-    const currentVal = await deployments.read(contract, options, getFunc)
-    if (currentVal !== val) {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      console.log('calling', `${contract}.${setFunc}( ${val.toString()} )`)
-      await deployments.execute(contract, options, setFunc, val)
-    }
   }
-
-  const balance = await ethers.provider.getBalance(deployer)
-  console.log('deployer=', deployer, 'balance=', formatEther(balance.toString()))
-  // if (balance.isZero()) {
-  //   console.error(chalk.red(`deployer account ${deployer as string} doesn't have eth balance ${balance.toString()}`))
-  //   process.exit(1)
-  // }
-
-  function fatal (...params: any): never {
-    console.error(chalk.red('fatal:'), ...params)
-    process.exit(1)
+  const currentVal = await deployments.read(contract, options, getFunc)
+  if (currentVal !== val) {
+    console.log('calling', `${contract}.${setFunc}( ${val} )`)
+    await deployments.execute(contract, options, setFunc, val)
   }
+}
 
-  const chainId = parseInt(await getChainId())
+function printSampleEnvironment (environment: Environment, chainId: number): void {
+  const sampleEnv = merge(environment, {
+    deploymentConfiguration: {
+      paymasterDeposit: '0.1',
+      deployTestPaymaster: true,
+      minimumStakePerToken: { test: '0.5' }
+    }
+  })
+  console.log(chalk.red('No configuration found:'), '\nPlease add the following to', chalk.whiteBright(`"${deploymentConfigFile}"`), ':\n\nmodule.exports = ',
+    util.inspect({ [chainId]: sampleEnv }, false, 10, false))
+}
 
-  let env: Environment
+function getMergedEnvironment (chainId: number): Environment {
   try {
-    const envname = Object.keys(environments).find(env => environments[env as EnvironmentsKeys].chainId === chainId)
-    console.log('loading env ( based on chainId', chainId, ')', envname ?? 'DefaultEnvironment')
-    const defaultEnv = envname != null ? environments[envname as EnvironmentsKeys] : defaultEnvironment
+    const env = getEnvironment(chainId)
+    if (env == null) {
```

Comment:
this will fail for any environment not already defined in our code.
the idea that for "goerli/rinkeby/fuji" (or any other "simple" network that doesn't require heavy fiddling with params) we only need to define a deployment in deployment-config.ts, not in the "Environments"